src/io.c

```diff
@@ -161,7 +170,11 @@ static void *pppd_read(void *arg)
 			continue;
 		} else if (first_time) {
 			// pppd did talk, now we can write to it if we want
+#ifndef __APPLE__
 			sem_post(&sem_pppd_ready);
+#else
+			dispatch_semaphore_signal(sem_pppd_ready);
+#endif
```

Comment:
Could we avoid all these `#ifndef __APPLE__` by defining a macro, like:

```c
#ifdef __APPLE__
#undef sem_post
#define sem_post(s) semaphore_signal(*s)
...
```
?

PS: Does defining the `USE_POSIX_SEM` constant provide `sem_post()`?